src/PROPOSAL/PROPOSAL/crosssection/CrossSection.h

```diff
@@ -228,9 +229,20 @@ class CrossSection : public CrossSectionBase {
     std::unique_ptr<std::vector<std::tuple<double, de2dx_ptr>>> de2dx;
 
     double lower_energy_lim;
+    double min_stochastic_energy;
     InteractionType interaction_type;
     std::string param_name;
 
+private:
+    template <typename T> static double CalculateMinStochasticEnergy(T const& m)
```

Comment:
Does this need to be a template? Isn't T always the same type?

Reply:
Not really, but I was to lazy to search the corresponding type :sweat_smile: